### PR TITLE
Fix issue #283: Rename Key Group to Entity Set

### LINKED ISSUE #283: Rename Key Group to Entity Set
It seems like we could adopt BIDS's terminology by referring to Key Groups (which are unique entity sets) as Entity Sets instead.

### PR DESCRIPTION
**Pull request by tientong98**
Closes #283.  <-- Taylor: put the issue you're addressing here so merging the PR will close the issue

## Changes proposed in this pull request

- Use re to change all instances of Key Group to Entity Set (matching capitalization and delimiter) across all files that can be decoded using UTF-8.
- Update to Ubuntu 2404:2024.08.1 and bids-validator@1.14.14-dev.0 to pass CircleCI test (from 1.14.9 to 1.14.13, there is an issue with [JSON schema validation](https://github.com/bids-standard/bids-validator/pull/2148)), change build_validator_call func in cubids/validator.py to follow the new syntax (require this version somewhere in the doc?)

### CODE REVIEW

**Comment by tientong98** [comment]
Hi @tsalo and @mattcieslak - circleci failed because it used ubuntu-2004:202201-02, which is now [deprecated and not available ](https://discuss.circleci.com/t/linux-image-deprecations-and-eol-for-2024/50177). I want to change the config.yml file so that it uses one of the [current images](https://discuss.circleci.com/t/ubuntu-20-04-22-04-24-04-q3-current-release/51856), but I'm not sure if it will break anything else. Do you have any preference on which image to use?

**Comment by mattcieslak** [comment]
Any of those should be ok

File: HISTORY.rst [hunk 1]
```diff
@@ -45,7 +45,7 @@ New Contributors
 * updated _update_json to no longer use pybids by @scovitz in https://github.com/PennLINC/CuBIDS/pull/232
 * Minor tune ups: codespell'ing (fixes + tox + CI (github actions)), remove of unintended to be committed 2 files by @yarikoptic in https://github.com/PennLINC/CuBIDS/pull/239
 * ENH: Make "NumVolumes" an integer for 3D images by @cookpa in https://github.com/PennLINC/CuBIDS/pull/211
-* adding note about fmap renamekeygroups by @megardn in https://github.com/PennLINC/CuBIDS/pull/140
+* adding note about fmap renameentitysets by @megardn in https://github.com/PennLINC/CuBIDS/pull/140
```
Comment (tsalo) [inline HISTORY.rst:48]: We shouldn't modify the changelog.

**Review by tsalo — CHANGES_REQUESTED** [review]
Just one requested change. Also, this is a breaking change, so I'd add a `breaking-change` label to the PR and we need to make sure the next release is a major one.

**Review by tsalo — APPROVED** [review]
LGTM!

File: docs/usage.rst [hunk 2]
```diff
@@ -10,49 +10,49 @@ More definitions
 ----------------
 
 
-.. _keygroup:
+.. _entityset:
 
-Key Group
+Entity Set
 ~~~~~~~~~
```
Comment (mattcieslak) [inline docs/usage.rst:16]:
```suggestion
~~~~~~~~~~~
```

**Review by mattcieslak — APPROVED** [review]
Excellent!! I like the new name much better

**Pull request merged**